### PR TITLE
fix: misleading hex parsing error message

### LINKED ISSUE #365: Bug: misleading hex parsing error message
Currently `HexParseError::InvalidLength` always assumes an expected length of `64`:

https://github.com/eqlabs/pathfinder/blob/24c307f0fe050248287c3df95e6285b23267de80/crates/stark_hash/src/hash.rs#L336

which is not true. I found out about this when getting this confusing error message:

`More than 64 digits found: 64 at line 1 column 336`

### PR DESCRIPTION
**Pull request by xJonathanLEI**
Fixes #365.

### CODE REVIEW

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Review by kkovaacs — APPROVED** [review]
Looks good to me. Thanks @xJonathanLEI!

**Comment by kkovaacs** [comment]
(Ignore the failing tests, those are caused by your PR not having access to Github workflow secrets.)

**Review by koivunej — APPROVED** [review]

File: crates/pathfinder/src/rpc/serde.rs [hunk 1]
```diff
@@ -567,7 +570,10 @@ mod tests {
         );
         assert_eq!(
             bytes_from_hex_str::<32>(TOO_LONG_HEX_STR),
-            Err(HexParseError::InvalidLength(65))
+            Err(HexParseError::InvalidLength {
+                max: 64,
+                actual: 65
+            })
         );
     }
```
Comment (CHr15F0x) [inline crates/pathfinder/src/rpc/serde.rs:578]:
```suggestion
        // Regression: previously max in the error message was hard-coded at 64,
        // so try another buf size to make sure it is not anymore
        assert_eq!(
            &format!("{}", bytes_from_hex_str::<1>("abc").unwrap_err()),
            "More than 2 digits found: 3"
        );
    }
```

**Review by CHr15F0x — APPROVED** [review]
LGTM, maybe a tiny regression test to make sure the bug is no more ;)

File: crates/pathfinder/src/rpc/serde.rs (hunk 1 above)
Comment (xJonathanLEI) [reply crates/pathfinder/src/rpc/serde.rs:578]: Oops already merged lol...

Comment (Mirko-von-Leipzig) [reply crates/pathfinder/src/rpc/serde.rs:578]: Yeah my bad.

Comment (CHr15F0x) [reply crates/pathfinder/src/rpc/serde.rs:578]: No problemo, it's already shipped, I can add this in a new micro pr ;) .

**Pull request merged**